flecsi/data/data_client_handle.h

```diff
@@ -50,7 +50,7 @@ struct data_client_handle_base_u : public DATA_CLIENT_TYPE,
   data_client_handle_base_u(const data_client_handle_base_u<DATA_CLIENT_TYPE,
     UNMAPPED_PERMISSIONS,
     DATA_POLICY> & h)
-    : DATA_POLICY(h), DATA_CLIENT_TYPE(h), type_hash(h.type_hash),
+    : DATA_CLIENT_TYPE(h), DATA_POLICY(h), type_hash(h.type_hash),
```

Comment:
Lots of changes like this one seem to be avoiding warnings that have nothing to do with HPX.  They do appear to be in their own commit, but it would really be nicer to have them in their own merge request (so that we can tell what's HPX-specific more easily).

Reply:
Correct, those warnings were produced by different compilers (partly MSVC, but also clang) and mostly are unrelated to the HPX backend. I will try separating those into their own merge request, however.

Reply:
See #645 for all changes unrelated to HPX.